### DISCUSSION TITLE
type string interfaces and displays should be available to type text as well

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#6072] [Feature Requests]
Many times a database field with max chars (so of type text for directus) is meant to have "type string" interfaces, like the simple input field. Since string is a subset of text, I'd offer string's displays and interfaces for a text field

**Comment by casey-chow** [comment]
As an quick fix, I've been handling this by going into the `directus_fields` table and manually setting the field interface to `input`. Works totally fine.

**Comment by rijkvanzanten** [comment]
Wanna add a PR to support `text` in the interfaces where it'd make sense? They're configured in each interface's `index.ts` file, for example: https://github.com/directus/directus/blob/ae65918d1e32ff4754a77e0fb63cecdee6be2f9d/app/src/interfaces/input/index.ts#L11

**Reply by wc-matteo** [reply]
sure, tomorrow I'll do that

**Reply by nazariydj** [reply]
I think varchar (string) can be textarea (input-multiline) too...

**Reply by rijkvanzanten** [reply]
That's where it becomes tricky.. A varchar has a limited number of characters available (by design) (historically 255 or less), so once you open up interfaces that save bigger chunks of data, like a wysiwyg or textarea, it becomes way easier for the end user to accidentally overload the column in the database 🤔 

Allowing varchar interfaces to be used with text makes sense, cause it's always "safe". The other way around not so much

**Reply by benhaynes** [reply]
I agree with this, but varchar (at least in MySQL) can go up to 65,535 characters (255 is just the default). So while it is certainty easier for admins to mess things up, I think that we should open up varchar to at least some (or all) of the text types.

Since the new Field Flow will typically choose an interface first, we _could_ increase the string length default for certain interfaces (eg: textarea).

**Reply by nazariydj** [reply]
In my case for seo description I need textarea with max 160 characters. It is very inconvenient to use one line for texts that have more than 50-60 characters. Perhaps the best solution may be able to set maxlength in text-multiline interface or use varchar size as textarea maxlength attribute.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
